homeassistant/components/heos/__init__.py

```diff
@@ -210,3 +214,66 @@ def connect_update(self, hass, controller):
 
         controller.dispatcher.connect(
             const.SIGNAL_CONTROLLER_EVENT, update_sources)
+
+
+class HeosRegistry:
```

Comment:
Ugh, it sucks that we can't have this be solved with our current infra :( Should we consider having some "unique ID migration" in entity registry ?

Reply:
I was thinking the same thing.  I think we'd need APIs `async_update_identifiers` and `async_update_unique_id` for the device and entity registries respectfully.  (The device registry merge functionality doesn't take care of _removing_ old IDs, which _could_ be reused in the future by HEOS.) . Adding the APIs themselves is not a big deal but I didn't have time to research how they were used throughout HA.  If the IDs are already decoupled, I can add those APIs tonight and ditch this PR.

Reply:
Writing to the entity and device registries only happens inside entity platform, invoked via `async_add_entities`.

Reply:
I'm currently reviewing another PR for ps4 where we need to migrate unique_id.

Reply:
I'll open an arch issue to discuss these changes.  My question isn't about when it's written, but if/how it's _used_ through HA.  For example, is unique_id used by other things, like the Recorder, that if it changes needs to be migrated there as well?

Reply:
Oh right. It currently is only used by the entity registry. I think that by design, nothing else should use it. Instead, they should use the entity registry ID instead.